pkg/cnb/create_builder.go

```diff
@@ -1,53 +1,59 @@
 package cnb
 
 import (
-	"github.com/google/go-containerregistry/pkg/authn"
-	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"context"
 
+	"github.com/google/go-containerregistry/pkg/authn"
+	ggcrv1 "github.com/google/go-containerregistry/pkg/v1"
 	buildapi "github.com/pivotal/kpack/pkg/apis/build/v1alpha2"
 	corev1alpha1 "github.com/pivotal/kpack/pkg/apis/core/v1alpha1"
+	"github.com/pivotal/kpack/pkg/registry"
+	k8scorev1 "k8s.io/api/core/v1"
 )
 
 type RegistryClient interface {
-	Fetch(keychain authn.Keychain, repoName string) (v1.Image, string, error)
-	Save(keychain authn.Keychain, tag string, image v1.Image) (string, error)
-}
-
-type BuildpackRepository interface {
-	FindByIdAndVersion(id, version string) (RemoteBuildpackInfo, error)
+	Fetch(keychain authn.Keychain, repoName string) (ggcrv1.Image, string, error)
+	Save(keychain authn.Keychain, tag string, image ggcrv1.Image) (string, error)
 }
 
 type LifecycleProvider interface {
-	LayerForOS(os string) (v1.Layer, LifecycleMetadata, error)
+	LayerForOS(os string) (ggcrv1.Layer, LifecycleMetadata, error)
 }
 
-type NewBuildpackRepository func(clusterStore *buildapi.ClusterStore) BuildpackRepository
+type BuilderCreator interface {
```

Comment:
Is this interface necessary?

Reply:
It's used in the builder reconcilers to inject a fake during tests
https://github.com/pivotal/kpack/blob/2bf41f99a971be7b171032c016b369a7de6ea27c/pkg/reconciler/builder/builder.go#L96-L99
https://github.com/pivotal/kpack/blob/2bf41f99a971be7b171032c016b369a7de6ea27c/pkg/reconciler/builder/builder_test.go#L44-L57